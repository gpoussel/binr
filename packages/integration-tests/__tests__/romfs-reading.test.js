"use strict";

const fs = require("fs");
const { BinaryReader } = require("@binr/binary-reader");
const { DefinitionReader } = require("@binr/definition-reader");

const pathToBinaryFixtures = `${__dirname}/../__fixtures__/binaries`;
const pathToDefinitionFixtures = `${__dirname}/../__fixtures__/definitions`;

describe("ROMFS reading", () => {
  const definitionReader = new DefinitionReader();
  const romfsDefinitionFile = fs.readFileSync(`${pathToDefinitionFixtures}/romfs.binr`);
  const romfsDefinition = definitionReader.readInput(romfsDefinitionFile.toString());

  const sampleRomfsFile = `${pathToBinaryFixtures}/sample.romfs`;
  const sampleRomfsBuffer = fs.readFileSync(sampleRomfsFile);

  test("reads ROMFS file", () => {
    const binaryReader = new BinaryReader();
    const value = binaryReader.read(sampleRomfsBuffer, romfsDefinition, "RomfsFile");
    expect(value).toBeDefined();
    const { header, entry } = value;
    expect(header).toBeDefined();
    expect(header.magicBytes).toEqual("-rom1fs-".split(""));
    expect(header.size).toBe(272);
    expect(header.checksum).toBe(1391339101);
    expect(header.volumeName).toEqual("DEFALIGNED");

    expect(entry).toBeDefined();

    const { header: entryHeader } = entry;
    expect(entryHeader).toBeDefined();
  });
});
