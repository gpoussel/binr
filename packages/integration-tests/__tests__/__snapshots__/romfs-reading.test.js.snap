// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ROMFS reading reads ROMFS file 1`] = `
Object {
  "entry": Object {
    "header": Object {},
  },
  "header": Object {
    "checksum": 1391339101,
    "magicBytes": "-rom1fs-",
    "size": 272,
    "volumeName": "DEFALIGNED",
  },
}
`;
